plugins/module_utils/vm.py

```diff
@@ -1410,20 +1585,31 @@ def send_delete_nic_request_to_hypercore(
             endpoint="/rest/v1/VirDomainNetDevice/" + nic_to_delete.uuid,
             check_mode=False,
         )
-        after.append(None)
         TaskTag.wait_task(rest_client=rest_client, task=response)
-        return True, before, after, virtual_machine_obj.reboot
+        after.append(None)
+        return True, before, after
 
     @classmethod
-    def ensure_present_or_set(cls, module, rest_client, module_path):
+    def ensure_present_or_set(cls, module, rest_client, module_path, vm_before: VM):
+        """
+
+        Args:
+            module:
+            rest_client:
+            module_path:
+            vm_before: The VM matching vm_name/vm_name_new
+
+        Returns:
+            changed:
+            before:
+            after:
+            reboot: bool, it means reboot required if VM was running before.
```

Comment:
we don't return reboot anymore right?

Reply:
The same info is now stored in the VM instance - it is returned by the `vm_before.was_vm_rebooted()`.